GeneratorInterface/Core/plugins/ExternalGeneratorFilter.cc

```diff
@@ -46,15 +46,22 @@ namespace externalgen {
 
       channel_.setupWorker([&]() {
         using namespace std::string_literals;
+        using namespace std::filesystem;
         edm::LogSystem("ExternalProcess") << id_ << " starting external process \n";
         std::string verboseCommand;
         if (verbose) {
           verboseCommand = "--verbose ";
         }
+        std::error_code ec;
+        auto curDir = current_path();
+        auto newDir = path("thread"s + std::to_string(id_));
+        create_directory(newDir, ec);
+        current_path(newDir, ec);
```

Comment:
I'm concerned that any possible errors are silently ignored.

Reply:
Thanks, shall this work?
```suggestion
        current_path(newDir, ec);
        if (ec) {
          throw cms::Exception("ExternalFailed")
            << "failed entering the working dir for process " << id_ << ": " << ec.message();
        }
```
(Actually the previous code was following this example https://github.com/cms-sw/cmssw/blob/c5d26737f2e7f2a28d2e485b18f07da4d4d79502/GeneratorInterface/LHEInterface/plugins/ExternalLHEProducer.cc#L481-L483)

Reply:
How about letting these functions to throw the exceptions? (i.e. remove the `ec` altogether, assuming the messages of those exceptions would be clear-enough)